frontend/src/components/UserProfile/Activities.tsx

```diff
@@ -27,34 +25,71 @@ const Activities = (props: Props) => {
         borderColor: 'primary.main',
       }}
     >
-      {props.activities.map((act) => {
-        const [doer, action, recipient] = JSON.parse(act.message);
-        return (
-          <>
-            <ListItem alignItems="flex-start" key={act.id}>
-              <ListItemAvatar>
-                <Avatar alt={act.user.name} src={act.user.avatar} />
-              </ListItemAvatar>
-              <ListItemText
-                primary={`${doer} ${action} ${recipient}`}
-                secondary={
-                  <React.Fragment>
-                    <Typography
-                      sx={{ display: 'inline' }}
-                      component="span"
-                      variant="body2"
-                      color="text.primary"
-                    >
-                      Ali Connors
-                    </Typography>
-                    {' — x time ago…'}
-                  </React.Fragment>
-                }
-              />
-            </ListItem>
-          </>
-        );
-      })}
+      {activities?.length === 0 || !activities
+        ? Array.from(Array(10).keys()).map((tempKey) => {
+            return (
+              <ListItem alignItems="flex-start" key={tempKey}>
+                <ListItemAvatar>
+                  <Skeleton variant="circular" width={40} height={40} />
+                </ListItemAvatar>
+                <Stack direction="column" spacing={2}>
+                  <Skeleton variant="rectangular" width={250} height={10} />
+                  <Skeleton variant="rectangular" width={250} height={10} />
+                </Stack>
+              </ListItem>
+            );
+          })
+        : activities?.map((act) => {
+            if (!act.log) return;
+            const [doer, action, recipient] = JSON.parse(act?.log?.message);
+            const date = new Date(act.created_at);
+            return (
+              <ListItem alignItems="flex-start" key={act.id}>
+                <ListItemAvatar>
+                  {act.following ? (
+                    <Avatar
+                      alt={act.following?.name}
+                      src={
+                        act.following?.avatar ??
+                        `${process.env.REACT_APP_BACKEND_URL}/storage/avatars/default.jpg`
+                      }
+                    />
+                  ) : (
+                    <Avatar
+                      alt={act.follower?.name}
+                      src={
+                        act.follower?.avatar ??
+                        `${process.env.REACT_APP_BACKEND_URL}/storage/avatars/default.jpg`
+                      }
+                    />
+                  )}
```

Comment:
You can use 1 component and do the condition inside the src and alt instead using 2 component

Reply:
done